### PR TITLE
Pass in uuid in blob property before construct the blobId

### PR DESCRIPTION
**Pull request by SophieGuo410**

### CODE REVIEW

File: ambry-router/src/main/java/com/github/ambry/router/PutOperation.java [hunk 1]
```diff
@@ -1350,10 +1350,17 @@ private void prepareForSending() {
           chunkBlobId = metadataPutChunk.reservedMetadataChunkId;
         } else {
           partitionId = getPartitionForPut(partitionClass, attemptedPartitionIds);
-          chunkBlobId =
-              new BlobId(routerConfig.routerBlobidCurrentVersion, BlobIdType.NATIVE, clusterMap.getLocalDatacenterId(),
-                  passedInBlobProperties.getAccountId(), passedInBlobProperties.getContainerId(), partitionId,
-                  passedInBlobProperties.isEncrypted(), blobDataType);
+          if (passedInBlobProperties.getReservedUuid() != null) {
```
[the remark this answers is not in the document]
Comment (SophieGuo410) [reply ambry-router/src/main/java/com/github/ambry/router/PutOperation.java]: You are right, thx for catching up this! Removed.

File: ambry-router/src/main/java/com/github/ambry/router/PutOperation.java [hunk 2]
```diff
@@ -1365,7 +1372,7 @@ private void prepareForSending() {
             passedInBlobProperties.getCreationTimeInMs(), passedInBlobProperties.getAccountId(),
             passedInBlobProperties.getContainerId(), passedInBlobProperties.isEncrypted(),
             passedInBlobProperties.getExternalAssetTag(), passedInBlobProperties.getContentEncoding(),
-            passedInBlobProperties.getFilename(), resolveReservedMetadataId());
+            passedInBlobProperties.getFilename(), resolveReservedMetadataId(), passedInBlobProperties.getReservedUuid());
```
[the remark this answers is not in the document]
Comment (SophieGuo410) [reply ambry-router/src/main/java/com/github/ambry/router/PutOperation.java]: Sure, updated.

File: ambry-router/src/main/java/com/github/ambry/router/PutOperation.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (justinlin-linkedin) [reply ambry-router/src/main/java/com/github/ambry/router/PutOperation.java]: the chunkBlobId here is also used by simple blob. You need to differentiate the blob data type here. Then it's simple blob type, we use reserved uuid, when it's a data chunk, we don't use it.

[the remark this answers is not in the document]
Comment (justinlin-linkedin) [reply ambry-router/src/main/java/com/github/ambry/router/PutOperation.java]: And one more thing we need to do is to make sure in BlobId constructor, whenever the uuid str is null or empty, we throw an exception. We don't want to have empty uuid.

**Review by justinlin-linkedin — APPROVED** [review]
Most part looks good to me. Just a few minor comments.

**Review by justinlin-linkedin — APPROVED** [review]

**Review by Arun-LinkedIn — APPROVED** [review]
Looks good to me. Had one minor comment

**Pull request merged**